builder/cloudstack/step_keypair.go

```diff
@@ -0,0 +1,141 @@
+package cloudstack
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"runtime"
+
+	"github.com/hashicorp/packer/packer"
+	"github.com/mitchellh/multistep"
+	"github.com/xanzy/go-cloudstack/cloudstack"
+)
+
+type stepKeypair struct {
+	Debug            bool
+	SSHAgentAuth     bool
+	DebugKeyPath     string
+	TemporaryKeyPair string
+	KeyPair          string
+	PrivateKeyFile   string
+
+	doCleanup bool
+}
+
+func (s *stepKeypair) Run(state multistep.StateBag) multistep.StepAction {
+	ui := state.Get("ui").(packer.Ui)
+
+	if s.PrivateKeyFile != "" {
+		privateKeyBytes, err := ioutil.ReadFile(s.PrivateKeyFile)
+		if err != nil {
+			state.Put("error", fmt.Errorf(
+				"Error loading configured private key file: %s", err))
+			return multistep.ActionHalt
+		}
+
+		state.Put("keypair", s.KeyPair)
```

Comment:
Shouldn't we `ui.Say` something above this line as well, about what we'll be using?